File: kotlin/lightning_talk/src/main/java/me/bendoan/helloKotlin/Nulls.java
package me.bendoan.helloKotlin;

import java.util.Random;

public class Nulls {
    public static void main(String[] args) {
        User user = getUser();

        System.out.printf("Length of name is %s", user.getName().length());

        // should really be:

        if (user != null) {
            if (user.getName() != null) {
                System.out.printf("Len is %s", user.getName().length());
            }
        }
    }

    private static User getUser(){
        if (new Random().nextBoolean()){
            return new User("Ben", 22);
        } else {
            return null;
        }
    }
}

File: kotlin/lightning_talk/src/main/kotlin/me/bendoan/helloKotlin/nulls.kt
package me.bendoan.helloKotlin

import java.util.*

fun main(args: Array<String>){
    val user: User = getUser()
    println("Length of name is ${user.name.length}")






























    val user2 = getUser2()

    // println("Length of name is ${user2.name.length}")

    if (user2 != null) {
        println("Length of name is ${user2.name.length}")
    }

    println("Length of name is ${user2?.name?.length}")

    println("Length of name is ${user2?.name?.length ?: "<undefined>"}")

    // For NPE-lovers
    println("Length of name is ${user2!!.name.length}")
}


























fun getUser(): User {
    return User("Ben", 22)
}

fun getUser2(): UserJava? {
    return if (Random().nextBoolean()){
        UserJava("Ben", 22)
    } else {
        null
    }
}